src/main/scala/au/com/agiledigital/toolform/command/generate/GenerateCommand.scala

```diff
@@ -0,0 +1,31 @@
+package au.com.agiledigital.toolform.command.generate
+
+import java.nio.file.Path
+
+import au.com.agiledigital.toolform.app.ToolFormError
+import au.com.agiledigital.toolform.plugin.ToolFormCommandPlugin
+import au.com.agiledigital.toolform.reader.ProjectReader
+import au.com.agiledigital.toolform.command.generate.docker.GenerateDockerComposeV3.runGenerateDockerComposeV3
+import com.monovore.decline._
+import cats.implicits._
+
+/**
+  * The primary class for generating config files.
+  * You should use this class as it will automatically delegate to the relevant subtask according to the configuration
+  * parsed on the command line.
+  */
+class GenerateCommand() extends ToolFormCommandPlugin {
+
+  def command: Opts[Either[ToolFormError, String]] =
+    Opts.subcommand("generate", "generates config files for container orchestration.") {
+      (Opts.option[Path]("in-file", short = "i", metavar = "file", help = "the path to the project config file") |@|
+        Opts.option[Path]("out-file", short = "o", metavar = "file", help = "the path to output the generated file(s)") |@|
```

Comment:
Should validate that the directory containing the target file exists.

Reply:
This looks like it has been implemented to check whether the target *file* exists rather than the *directory* containing the target file exists. Doesn't that mean that you have to touch the output file before running the command?

Reply:
That's the way the code looks to me as well.
All I really did in this PR is to port over the existing docker generate function to the new CLI parser.  I didn't want to change any behaviour particularly.  Happy to fix it up in a separate PR though.

Reply:
Edit above: while porting things over you asked me to add the validation, so it is net new.

Sorry, been a long day here already :+)
The logic was haywire for output validation.  I'm working on a fix.